software/controller/lib/hal/stepper.h

```diff
@@ -119,48 +115,46 @@ enum class StepMtrErr {
   InvalidState,  // Invalid state for command.
 };
 
-enum class StepMoveStatus {
-  Stopped,
-  Accelerating,
-  Decelerating,
-  ConstantSpeed,
+enum class MoveStatus {
+  Stopped = 0,
+  Accelerating = 1,
+  Decelerating = 2,
+  ConstantSpeed = 3,
 };
 
 // Detailed status about the stepper driver chip.
 // Note, latching fields will return true if the event has occurred since the last time
 // the status was read.  The act of reading the status clears them so they won't be true
 // on the next read unless they happen again.
-struct StepperStatus {
-  bool enabled{false};
-  bool under_voltage{false};     // latching
-  bool thermal_warning{false};   // latching
-  bool thermal_shutdown{false};  // latching
-  bool over_current{false};      // latching
-  bool step_loss{false};         // latching
-  bool command_error{false};     // latching
-  StepMoveStatus move_status{StepMoveStatus::Stopped};
+union Status {
+  struct {
+    uint16_t disabled : 1;
```

Comment:
Would `uin8_t` not suffice if it's maximum of 2 bits from what I see? Or does it not matter?